File: petpals-pals/src/main/java/com/petpals/bootstrap/interceptors/ExceptionHandler.java
package com.petpals.bootstrap.interceptors;

import com.petpals.shared.errorhandling.PetPalsExceptions;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.ext.ExceptionMapper;
import jakarta.ws.rs.ext.Provider;

@Provider
public class ExceptionHandler implements ExceptionMapper<PetPalsExceptions> {
    @Override
    public Response toResponse(PetPalsExceptions e) {
        return Response.status(Response.Status.fromStatusCode(e.getAppError().getHttpResponseStatus())).entity(e.getAppError().getMessage()).build();
    }
}

File: petpals-pals/src/main/kotlin/com/petpals/bootstrap/ExceptionHandler.kt
package com.petpals.bootstrap;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.petpals.shared.errorhandling.PetPalsExceptions;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.ext.ExceptionMapper
import org.jboss.logging.Logger;


class ExceptionHandler : ExceptionMapper<PetPalsExceptions> {
    override fun toResponse(e: PetPalsExceptions): Response {
        val objectMapper = ObjectMapper()
        return try {
            Logger.getLogger("here").info(objectMapper.writeValueAsString(e))
            Response.status(Response.Status.fromStatusCode(e.appError.httpResponseStatus))
                    .entity(objectMapper.writeValueAsString(e.appError))
                    .build()
        } catch (ex: JsonProcessingException) {
            Response.status(Response.Status.fromStatusCode(e.appError.httpResponseStatus))
                    .entity(e.appError.message)
                    .build()
        }
    }
}